src/jit.rs

```diff
@@ -461,35 +462,51 @@ fn emit_bpf_call(jit: &mut JitMemory, dst: Value, number_of_instructions: usize,
+            // Force alignment of RAX
+            emit_alu64_imm32(jit, 0x81, 4, REGISTER_MAP[0], !(INSN_SIZE as i32 - 1)); // RAX &= !(INSN_SIZE - 1);
             // Store PC in case the bounds check fails
             emit_load_imm(jit, R11, pc as i64 + ebpf::ELF_INSN_DUMP_OFFSET as i64);
-            // Force alignment of RAX
-            emit_alu64_imm32(jit, 0x81, 4, RAX, !7); // RAX &= !(8 - 1);
             // Upper bound check
-            // if(RAX > MM_PROGRAM_START + (number_of_instructions - 1) * INSN_SIZE) throw CALL_OUTSIDE_TEXT_SEGMENT;
-            emit_load_imm(jit, RBX, MM_PROGRAM_START as i64 + ((number_of_instructions - 1) * INSN_SIZE) as i64);
-            emit_cmp(jit, RBX, RAX);
-            emit_jcc(jit, 0x87, TARGET_PC_CALL_OUTSIDE_TEXT_SEGMENT);
+            // if(RAX >= MM_PROGRAM_START + number_of_instructions * INSN_SIZE) throw CALL_OUTSIDE_TEXT_SEGMENT;
+            emit_load_imm(jit, REGISTER_MAP[STACK_REG], MM_PROGRAM_START as i64 + (number_of_instructions * INSN_SIZE) as i64);
+            emit_cmp(jit, REGISTER_MAP[STACK_REG], REGISTER_MAP[0]);
+            emit_jcc(jit, 0x83, TARGET_PC_CALL_OUTSIDE_TEXT_SEGMENT);
             // Lower bound check
             // if(RAX < MM_PROGRAM_START) throw CALL_OUTSIDE_TEXT_SEGMENT;
-            emit_load_imm(jit, RBX, MM_PROGRAM_START as i64);
-            emit_cmp(jit, RBX, RAX);
+            emit_load_imm(jit, REGISTER_MAP[STACK_REG], MM_PROGRAM_START as i64);
+            emit_cmp(jit, REGISTER_MAP[STACK_REG], REGISTER_MAP[0]);
             emit_jcc(jit, 0x82, TARGET_PC_CALL_OUTSIDE_TEXT_SEGMENT);
             // Calculate offset relative to instruction_addresses
-            emit_alu64(jit, 0x29, RBX, RAX); // RAX -= MM_PROGRAM_START;
+            emit_alu64(jit, 0x29, REGISTER_MAP[STACK_REG], REGISTER_MAP[0]); // RAX -= MM_PROGRAM_START;
             // Load host target_address from JitProgramArgument.instruction_addresses
-            emit_mov(jit, R10, RBX);
-            emit_alu64(jit, 0x01, RBX, RAX); // RAX += &JitProgramArgument as *const _;
-            emit_pop(jit, RBX);
-            emit_load(jit, OperandSize::S64, RAX, RAX, std::mem::size_of::<MemoryMapping>() as i32); // RAX = JitProgramArgument.instruction_addresses[RAX / 8];
+            emit_mov(jit, R10, REGISTER_MAP[STACK_REG]);
+            emit_alu64(jit, 0x01, REGISTER_MAP[STACK_REG], REGISTER_MAP[0]); // RAX += &JitProgramArgument as *const _;
+            emit_load(jit, OperandSize::S64, REGISTER_MAP[0], REGISTER_MAP[0], std::mem::size_of::<MemoryMapping>() as i32); // RAX = JitProgramArgument.instruction_addresses[RAX / 8];
+        },
+        Value::Constant(_target_pc) => {},
+        _ => panic!()
+    }
+
+    emit_load(jit, OperandSize::S64, RBP, REGISTER_MAP[STACK_REG], -8 * CALLEE_SAVED_REGISTERS.len() as i32); // load stack_ptr
+    emit_alu64_imm32(jit, 0x81, 4, REGISTER_MAP[STACK_REG], !(CALL_FRAME_SIZE as i32 * 2 - 1)); // stack_ptr &= !(CALL_FRAME_SIZE * 2 - 1);
```

Comment:
Be nice if these x86 opcodes weren't magic numbers, though not hard to go look up

Reply:
Yes, I have been thinking about using enums or consts, but that would be its own refactoring PR.